tidy3d/web/api/mode.py

```diff
@@ -425,17 +426,28 @@ def get_result(
             Mode solver data with the calculated results.
         """
         try:
-            download_file(
-                self.solver_id,
-                MODESOLVER_RESULT,
+            download_gz_file(
+                resource_id=self.solver_id,
+                remote_filename=MODESOLVER_RESULT_GZ,
                 to_file=to_file,
                 verbose=verbose,
                 progress_callback=progress_callback,
             )
         except Exception:
-            raise WebError(
-                f"Failed to download file '{MODESOLVER_RESULT}' from server. Please confirm that the task was successful."
-            )
+            log.debug("Unable to download compressed solver data file. Trying uncompressed.")
+            try:
+                download_file(
+                    resource_id=self.solver_id,
+                    remote_filename=MODESOLVER_RESULT,
+                    to_file=to_file,
+                    verbose=verbose,
+                    progress_callback=progress_callback,
+                )
+            except Exception:
+                raise WebError(
```

Comment:
probably best to catch the specific exception here, and then raise `from e`. That way more info gets displayed to the user if this occurs so we can debug it?